kea-vscode/src/commands/commands/open-commit-file-diff.ts

```diff
@@ -0,0 +1,175 @@
+      vscode.window.showErrorMessage("Failed to open file diff");
+    }
+  };
+
+async function handleLocalCommitFileDiff(commitSha: string, filePath: string, workspacePath: string): Promise<void> {
+  try {
+    // Create a LocalGitRepository instance.
+    const localGitRepo = new LocalGitRepository(workspacePath, new ApiCache(100));
+    
+    // Get the parent commit to show what changed in this commit.
+    const parentCommit = await localGitRepo.getParentCommit(commitSha);
+    
+    let leftContent: string;
+    let leftTitle: string;
+    
+    if (parentCommit instanceof Error) {
+      // This might be the first commit, show empty content.
+      leftContent = "";
+      leftTitle = "Initial";
+    } else {
+      // Get file content at parent commit.
+      const parentContent = await localGitRepo.getFileAtCommit(parentCommit, filePath);
+      if (parentContent instanceof Error) {
+        // File didn't exist in parent commit, show empty content.
+        leftContent = "";
+        leftTitle = "New file";
+      } else {
+        leftContent = parentContent;
+        leftTitle = parentCommit.substring(0, 7);
+      }
+    }
+    
+    // Get file content at the commit we're viewing.
+    const commitContent = await localGitRepo.getFileAtCommit(commitSha, filePath);
+    
+    if (commitContent instanceof Error) {
+      Logger.error(`Failed to get commit content for ${filePath} at ${commitSha}`, commitContent);
+      vscode.window.showErrorMessage(`Failed to read file at commit ${commitSha}: ${commitContent.message}`);
+      return;
+    }
+
+    // Create temporary files with proper extensions for syntax highlighting.
+    const tempDir = path.join(os.tmpdir(), 'kea-commit-files');
+    
+    try {
+      await fs.mkdir(tempDir, { recursive: true });
+    } catch {
+      // Directory might already exist, ignore error.
+    }
+    
+    const fileExtension = path.extname(filePath);
+    const baseName = path.basename(filePath, fileExtension);
+    
+    const leftTempFileName = `${baseName}.${leftTitle}${fileExtension}`;
+    const rightTempFileName = `${baseName}.${commitSha.substring(0, 7)}${fileExtension}`;
+    
+    const leftTempFilePath = path.join(tempDir, leftTempFileName);
+    const rightTempFilePath = path.join(tempDir, rightTempFileName);
+    
+    await fs.writeFile(leftTempFilePath, leftContent, 'utf8');
+    await fs.writeFile(rightTempFilePath, commitContent, 'utf8');
+    
+    const leftTempFileUri = vscode.Uri.file(leftTempFilePath);
+    const rightTempFileUri = vscode.Uri.file(rightTempFilePath);
+    
+    // Open diff editor with parent commit on the left and current commit on the right.
+    // Show the full file path in the title to indicate which file is being diffed.
+    await vscode.commands.executeCommand(
+      "vscode.diff",
+      leftTempFileUri,
+      rightTempFileUri,
+      `${filePath} (${leftTitle} ↔ ${commitSha.substring(0, 7)})`,
+      {
+        preview: true,
+      }
+    );
```

Comment:
The path for the LHS of the file is still in the local temp directory. It should show the path of the file in the file system, if the repo was checked out at that point.

i.e., instead of `C:\Users\DALYIS~1\AppData\Local\Temp\kea-commit-files\commits-root-tree-node.06560b7.ts`, it should be in `C:\Users\dalyisaac\Repos\kea\kea-vscode\src\views\pull-request-contents\commits\commits-root-tree-node.ts`